### PR TITLE
consensus/ethash, core: implement Metropolis EIP 100

### PR DESCRIPTION
**Pull request by karalabe**
Implements https://github.com/ethereum/EIPs/issues/100.

### CODE REVIEW

File: core/chain_makers.go [hunk 1]
```diff
@@ -218,6 +218,7 @@ func makeHeader(config *params.ChainConfig, parent *types.Block, state *state.St
 			Number:     parent.Number(),
 			Time:       new(big.Int).Sub(time, big.NewInt(10)),
 			Difficulty: parent.Difficulty(),
+			UncleHash:  parent.UncleHash(),
```
Comment (holiman) [inline core/chain_makers.go]: What is the effect of this change ?

Comment (karalabe) [inline core/chain_makers.go]: This is only used in our tests when we create artificial chains to test various corner cases. The chain maker is messing around with incomplete blocks/headers to get around certain validity checks (to evoke failures), so here we don't have a "real" parent block to calculate the difficulty of the child's based on, but instead make one up on the fly. The difficulty algo didn't require access to the uncles until now, so we just skipped copying that field over. It's however needed in metro.
